docker/docker_client.go

```diff
@@ -363,6 +363,8 @@ func SearchRegistry(ctx context.Context, sys *types.SystemContext, registry, ima
 	hostname := registry
 	if registry == dockerHostname {
 		hostname = dockerV1Hostname
+		// On docker.io if image is "library/foo", it's stored under "foo"
```

Comment:
(I realize this comment is motivated by the Docker one, still…)

- I don’t know what “stored under” means here (the Docker code says “stored locally”, which is even more confusing). `docker.io/library/busybox` and `docker.io/busybox/busybox` would be different names, so it’s can not really be true that the `library` one is “stored under” the other one — and anyway where an image is stored on a registry an internal registry detail, ~irrelevant to invoking a search.
- What we _care_ about is that (as of today, Oct 2023) a search term of `library/foo` does not find the `library/foo` image on the `docker.io` servers, which is surprising - and that Docker is modifying the search term client-side this same way, and it seems convenient to do the same thing. So I think the comment should say something like that.